app/scripts/app.js

```diff
@@ -10,24 +10,43 @@ var app = angular.module('app', [
     'ngAnimate'
 ]);
 
-app.config(function ($routeProvider, $locationProvider) {
-    $routeProvider
-    .when('/', {
+window.routes = {
+    "/": {
         templateUrl: 'views/home.html',
-        controller: 'HomeCtrl'
-    })
-    .when('/login', {
+        controller: 'HomeCtrl',
+    },
+    "/login": {
         templateUrl: 'views/login.html',
-        controller: 'LoginCtrl'
-    })
-    .when('/cadastro', {
+        controller: 'LoginCtrl',
+        onlyGuest: true
+    },
+    "/cadastro": {
         templateUrl: 'views/register.html',
-        controller: 'RegisterCtrl'
-    })
-    .otherwise({
-        templateUrl: '404.html',
-        controller: 'ErrorCtrl'
-    }); 
+        controller: 'RegisterCtrl',
+        onlyGuest: true
+    }
+};
 
+app.config(function ($routeProvider, $locationProvider) {
+    for(var path in window.routes) {
+        $routeProvider.when(path, window.routes[path]);
+    }
+    $routeProvider.otherwise({redirectTo: '/404'});
     $locationProvider.html5Mode(true);
 });
+
+app.run(function($rootScope, $location, User){
+    $rootScope.$on("$locationChangeStart", function(event, next) {
+        for(var i in window.routes) {
+            if(next.indexOf(i) !== -1) {
+                if(window.routes[i].onlyLogged && !User.isLogged()) {
```

Comment:
A idéia é: se eu tenho o atributo `onlyLogged` significa que eu quero somente usuários logados naquela rota, logo, só faço a segunda validação se esse atributo for true, por que se eu não tenho esse atributo naquela rota que estou tentando acessar, pouco me importa se meu usuário está logado ou não.